delphi-frontend/src/main/java/au/com/integradev/delphi/symbol/scope/DelphiScopeImpl.java

```diff
@@ -170,6 +170,14 @@ private boolean isAcceptableDuplicate(NameDeclaration declaration) {
 
     Set<NameDeclaration> duplicates = declarationsByName.get(declaration.getImage());
 
+    // Unit imports can clash with other declarations, except other imports
+    if (declaration instanceof UnitImportNameDeclaration) {
+      return duplicates.stream().noneMatch(UnitImportNameDeclaration.class::isInstance);
+    } else {
+      // Disregard unit imports when checking for duplicates
+      duplicates.removeIf(UnitImportNameDeclaration.class::isInstance);
```

Comment:
`duplicates` is a mutable set returned by `TreeMultimap::get`.
Modifying it in any way will modify the multimap, so we should do this differently.

Perfect-world, in a way that doesn't involve making a copy of the set.

Reply:
I've refactored to store and modify the stream instead.